flocker/provision/_install.py

```diff
@@ -193,50 +204,87 @@ def configure_firewalld(rule):
                         ['firewall-cmd']]])
 
 
-def task_enable_flocker_control():
+def task_enable_flocker_control(distribution):
     """
     Enable flocker-control service.
     """
-    return sequence([
-        run_from_args(['systemctl', 'enable', 'flocker-control']),
-        run_from_args(['systemctl', 'start', 'flocker-control']),
-    ])
+    if distribution in ('centos-7', 'fedora-20'):
+        return sequence([
+            run_from_args(['systemctl', 'enable', 'flocker-control']),
+            run_from_args(['systemctl', 'start', 'flocker-control']),
+        ])
+    elif distribution == 'ubuntu-14.04':
+        # Since the flocker-control service is currently installed
+        # alongside the flocker-agent service, the default control
+        # service configuration does not automatically start the
+        # service.  Here, we provide an override file to start it.
+        return sequence([
+            put(
+                path='/etc/init/flocker-control.override',
+                content=dedent('''\
+                    start on runlevel [2345]
+                    stop on runlevel [016]
+                    '''),
+            ),
+            run_from_args(['service', 'flocker-control', 'start']),
+        ])
+    else:
+        raise NotImplementedError()
 
 
-def task_open_control_firewall():
+def task_open_control_firewall(distribution):
     """
     Open the firewall for flocker-control.
     """
-    return sequence([
-        configure_firewalld(['--add-service', service])
-        for service in ['flocker-control-api', 'flocker-control-agent']
-    ])
+    if distribution in ('centos-7', 'fedora-20'):
+        return sequence([
+            configure_firewalld(['--add-service', service])
+            for service in ['flocker-control-api', 'flocker-control-agent']
+        ])
+    elif distribution == 'ubuntu-14.04':
+        return sequence([])
```

Comment:
It's true that UFW doesn't appear to be enabled anywhere by default.  However, I'm adding some changes to set UFW rules on Ubuntu.  As long as UFW is installed, which seems to be the case for all but most minimal systems (e.g. Docker), the firewall commands will succeed even if the firewall is not enabled.